### ISSUE TITLE
Hi! Some issues

### ISSUE THREAD

**Issue by Rex000** [kubaska/twitch-fox#1]
Hey @kubaska,

Haven't heard from you in a while nor there was any new updates for TF. I thought you didn't read my last message so I thought about posting here.

A few more issues that I just came across:
- The refresh button is clickable in the games tab but I don't know if it actually works. I wish it was usable in the streams tab too as I've noticed that often times when I click on a stream, the twitch page opens up and shows that the stream already got offline a few minutes ago.

- When searching for a stream or game, the "exit search" button doesn't work. The only way is to either change the tab then come back or manually delete the search letters. I don't remember if this was the case before so I thought I should bring it up.

- I noticed that the thumbnails don't show the profile pic of the streamer anymore. IDK if it was intentional but it was a good thing to have. Now more often than not when I'm skimming through a large number of live streams, I find myself skipping streamers because they're not on the thumbnail(i.e. if they're playing a game for instance), and it's already hard to remember all of their names. It was very convenient from the profile pics to quickly know which streamer's thumbnail I was looking at, and thus it was not easy to miss them.

- I accidentally removed Twitch Fox while removing another addon. When I reinstalled it and loaded my list, it only had my followed channels but not the games. The followed games tab was completely empty.

Glad to hear that the game data can be displayed using GQL api. As I've stated before too that it's really hard to click on every single game thumbnail to check if there are any live streams for it. It's a shame that they made the api less useful than it used to be instead of the other way around. I really hope the GQL api works as it would be a very useful feature to have. I'm eagerly waiting for it.

Thanks!

**Comment by kubaska** [comment]
Hello @Rex000,
Sorry for my really late reply, and about not replying at all on your other comment. I was busy with work and life stuff and completely forgot about it.

About your reported issues:
- I have updated refresh button behavior, so now you'll be able to refresh content on "unrefreshable" tabs.

- I also updated exit search button behavior, to clear search query first, and then wipe search levels. I think you already asked for this earlier, but I didn't make it work well.

- As for streamer thumbnail photos, I really miss them too. It's one of Twitch's api downgrades, and the only way to get them is to send additional request, but this basically doubles the time for you to see content. Since I have followed channels cached, I can display thumbnails in followed streams or followed vods - I wanted to add this back in january, but didn't since I wanted to be consistent with design across the extension.

- Yes, followed games are not being saved in backup file. This is on my to-do list.

About GQL: I started reading about it and have mixed feelings. For example, one of the reasons Chatterino is not using it is that they don't want to risk a lawsuit. Though, for an extension this small it would probably be fine to implement it as a "proof of concept". The bigger issue is that when I tried using it in browser there was no way to send a request without cookies - this means if you're logged in to Twitch (in browser, not TF), they can connect request with your account, and apparently there is a risk that you can get banned for that.
So, as you see it's not that easy as just implementing it. Of course, there are ways to mitigate that, but it's falling outside the scope of a browser extension.

I'll see what can I do about it in the coming weeks. Thanks for using Twitch Fox.

**Comment by Rex000** [comment]
Hey @kubaska,

Wow, I can't believe how much they downgraded their api. First we couldn't have the game data, and now we can't even have the profile pics. 

Something off the top of my head about profile pics: Is it possible that we could make refreshing them a manual thing? Like for instance have a "Refresh all profile pics" button in the add-on options that only updates all of them if the user manually clicks on it? That would be enough I guess. Just a thought.

But regardless, thanks for the awesome work!

**Comment by Rex000** [comment]
v5.1.3 just got installed and I immediately noticed that the refresh button is still not removing offline streams from the list. The empty thumbnails remain there forever no matter how many times you refresh. I refreshed even after an hour and they were still not removed.

Games tab
![Screenshot - 06_09_2022 , 2_43_14 am](https://user-images.githubusercontent.com/28770012/188519237-5f807ee0-3591-4056-8b38-b590e3ef98c2.png)


Streams tab
![Screenshot - 06_09_2022 , 4_04_43 am](https://user-images.githubusercontent.com/28770012/188519246-6cc026ee-8744-4ff7-a279-267e4bec3efe.png)

**Comment by kubaska** [comment]
Just to confirm, are you sure you're refreshing with button from the addon, the one to the right of search bar and to the left of your Twitch avatar, and not the browser one?
I just tried refreshing after a while and it works just fine for me.

**Comment by Rex000** [comment]
Ah ok, you're right. I recently made a habit of having a tab open in extended view and I would just do F5 to refresh because it feels much easier. I thought it achieved the same thing as clicking the refresh button on the addon but apparently that's not the case. It would be nice though if we could refresh using F5 in extended view.

**Comment by Rex000** [comment]
Ah ok, you're right. I recently made a habit of having a tab open in extended view and I would just do F5 or click the browser refresh button to refresh because it felt much easier. I thought it achieved the same thing as clicking the refresh button on the addon but apparently that's not the case. It would be nice though if we could refresh using F5 and browser button too in extended view.

**Comment by Rex000** [comment]
Ah, you're right. I was indeed doing F5 or clicking the browser refresh button because that's what I do in my followed streams tab, since it doesn't have a refresh button. So naturally I was doing the same thing in other tabs too, but apparently that's not the case in other tabs. 

Though for consistency and convenience sake, I think it would be a good thing to have the F5/browser button refresh ability across all tabs.

**Comment by Rex000** [comment]
Ah, you're right. I was indeed doing F5 or clicking the browser refresh button, because that's what I do in my followed streams tab in extended view since it doesn't have a refresh button, and doing so actually works. So naturally I was doing the same thing in other tabs too, but apparently that's not the case with other tabs.

Though for consistency and convenience sake, I think it would be a good thing to have the F5/browser button refresh ability across all tabs, in extended view.

**Comment by Rex000** [comment]
Ah, you're right. I was indeed doing F5 or clicking the browser refresh button because that's what I do in my followed streams tab in extended view, and doing so actually works. So naturally I was doing the same thing in other tabs too, but apparently that's not the case with them.

Though for consistency and convenience sake, I think it would be a good thing to have the F5/browser refresh button ability across all tabs, in extended view.

**Comment by Rex000** [comment]
Hi, just noticed an inconsistency in the add-on's refresh button. It does remove the empty thumbnails as stated above but it's not updating the thumbnail images. Only clicking the browser's refresh button updates them.

**Comment by Rex000** [comment]
Hi, just noticed an inconsistency in the add-on's refresh button. It does remove the empty thumbnails as stated above but it often doesn't update the thumbnail images. Only clicking the browser's refresh button updates them.

**Comment by kubaska** [comment]
Hi Rex,
I've done some testing on this, and they're refresh fine for me. Stream previews are updated every so often, and Twitch servers tell your browser to cache them for 5 minutes. So it seems that in the worst case scenario thumbnails will not change for ~10 minutes? But I can confirm refresh button works. I am rerendering all stream cards after all.

**Comment by Rex000** [comment]
Yeah it does change the thumbnails if you wait that long. What I don't understand though is why it updates them once browser refresh is clicked, even if it's been under a minute?

Couple more things. Firefox recently updated and the streamer notification popup seems to now be receiving on Windows side instead of browser side. At least that's the case for Windows 10.  This is ok, but now when you click on the popup it randomly sometimes opens up the stream while other times it doesn't. I have my setting "After clicking notification" set to "Stream in new tab".

Moreover the notification popup doesn't appear for every stream that comes online. Seems like it randomly works.
